src/components/ImageModal/index.tsx

```diff
@@ -0,0 +1,351 @@
+import type { HTMLAttributes, ReactElement, TouchEventHandler } from 'react'
+import { useEffect, useMemo, useRef, useState } from 'react'
+import { IconButton } from '@components'
+import ReactDOM from 'react-dom'
+import styled from '@emotion/styled'
+import type { StyledProps } from '@types'
+
+interface ImageInfo {
+  src: string
+  id: string
+}
+export interface ImageModalProps extends HTMLAttributes<HTMLDivElement> {
+  isOpen?: boolean
+  images: ImageInfo[]
+  name: string
+  onClose?(): void
+}
+type ResizeImageInfo = ImageInfo & {
+  width: number
+  height: number
+}
+
+type StyledDIMProps = StyledProps<ImageModalProps, 'isOpen'>
+interface StyledImageModalProps {
+  isFixedHeight: boolean
+  direction: 'top' | 'bottom'
+}
+interface StyledImageContainerProps {
+  imageCount: number
+  imageWidth: number
+}
+type StyledImageProps = Pick<StyledImageModalProps, 'isFixedHeight'>
+type StyledGradientProps = Pick<StyledImageModalProps, 'direction'>
+
+const DEFAULT_RESIZE_HEIGHT = 640
+const IMAGE_GAP_HALF = 6
+const TOUCH_START_END_DIFFERENCE = 30
+
+export const ImageModal = ({
+  onClose,
+  images,
+  isOpen = false,
+  name,
+  ...props
+}: ImageModalProps): ReactElement => {
+  const imagesInfo = useRef<ResizeImageInfo[]>([])
+  const firstImageId = images[0].id
+  const [currentImageId, setCurrentImageId] = useState<string>(firstImageId)
+  const clientX = useRef<number | null>(null)
+  const topElement = useMemo(() => document.createElement('div'), [])
+
+  useEffect(() => {
+    document.body.append(topElement)
+
+    return () => {
+      document.body.removeChild(topElement)
+    }
+  }, [])
+
+  useEffect(() => {
+    getImagesInfo()
+  }, [images])
+
+  useEffect(() => {
+    setCurrentImageId(firstImageId)
+  }, [isOpen])
+
+  const getImagesInfo = async (): Promise<void> => {
+    const fulfilledImages = images.map(({ src, id }) => {
+      const image = new Image()
+      image.src = src
+
+      return new Promise(resolve => {
+        image.onload = (): void => {
+          resolve({
+            height: image.height,
+            id,
+            src,
+            width: image.width * (DEFAULT_RESIZE_HEIGHT / image.height)
+          })
+        }
+      })
+    })
+
+    imagesInfo.current = (await Promise.all(
+      fulfilledImages
+    )) as ResizeImageInfo[]
+  }
+
+  const getSumImageWidth = (): number =>
+    [...imagesInfo.current].reduce(
+      (sumImageWidth, currentImage, idx, images) => {
+        const { id, width } = currentImage
+
+        if (currentImageId === id) {
+          // for early return
+          images.splice(1)
+          return sumImageWidth + width / 2 + IMAGE_GAP_HALF
+        }
+
+        return sumImageWidth + width + IMAGE_GAP_HALF
+      },
+      0
+    )
+
+  const getImageCount = (): number => {
+    return imagesInfo.current.findIndex(({ id }) => currentImageId === id)
+  }
```

Comment:
```suggestion
  const getHandOverImageCount = (): number => {
    return imagesInfo.current.findIndex(({ id }) => currentImageId === id)
  }
```
넘겨진 이미지에 대한 개수여서 요 네이밍으로 변경하는 건 어떨까요!?

Reply:
[요기](https://github.com/price-offer/offer-design-system/pull/35/commits/e90f6d96972c4a90227154e46f0ae5ee2f7b2333)에 반영 완료 😘